fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/CollateFunctionKeyExpression.java

```diff
@@ -180,9 +179,13 @@ public int getColumnSize() {
 
     @Nonnull
     @Override
-    public Value toValue(@Nonnull final CorrelationIdentifier baseAlias,
-                         @Nonnull final Type baseType,
-                         @Nonnull final List<String> fieldNamePrefix) {
+    public <S extends KeyExpressionVisitor.State, R> R expand(@Nonnull final KeyExpressionVisitor<S, R> visitor) {
+        return visitor.visitExpression(this);
+    }
+
+    @Nonnull
+    @Override
+    public Value toValue(@Nonnull final List<? extends Value> argumentValues) {
         // TODO support this
```

Comment:
See https://github.com/FoundationDB/fdb-record-layer/pull/2899/commits/02185cf403edbba1e8d65b5802e77fe16980529a and comment below.

Reply:
That commit, if applied on top of this one, can now use `resolveAndEncapsulateFunction()`